pwnlib/heap/glmalloc/process_informer.py

```diff
@@ -0,0 +1,51 @@
+import os.path
+import pwnlib.util.proc
+from pwnlib.heap.glmalloc.utils import \
+    u64, \
+    u32, \
+    get_libc_version_from_name, \
+    get_main_arena_addr
+
+
+class ProcessInformer:
+    """Helper class with information about process memory such as pointer_size,
+    and which allows perform operations over memory.
+
+    Attributes:
+        libc_version (tuple(int, int)): The glibc version in
+            format (major, minor)
+        pid (int): pid of the process
+        pointer_size (int): size in bytes of a pointer in the process
+        main_arena_address (int): address of the main arena malloc state
+
+    """
+
+    def __init__(self, pid, libc):
```

Comment:
libc detection should be automatic, we can do this with Corefille

Reply:
Hi, what do you mean with automatic? 

On the other hand, I've implemented a CorefileInformer to get the heap information from a corefile. However it requires to parse the libc file in order to extract it symbols, since those are not processed in the corefile. Is there any way to get the libc symbols directly from a corefile?